cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -320,4 +340,25 @@ private void popStash(@Nullable RevCommit stashID) throws GitAPIException {
     LOG.warn("Couldn't find Git stash with ID {} to drop. Ignoring missing stash.", stashID.getName());
   }
 
+  /**
+   * Does a <a href="https://git-scm.com/docs/git-reset"> hard reset </a> to the remote branch being tracked.
+   */
+  private String resetToCleanBranch() throws IOException, GitAPIException {
```

Comment:
We need to fetch the latest heads also. This method is essentially a fetch followed by a rest hard. There a call to clean() so that any untracked and ignored files are also discarded and you get a clean slate.